### PR TITLE
Add dwarves package

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This change adds the dwarves package.

* dwarves: import dwarves from F35

* dwarves: add Mariner definitions

* cgmanifest: Add dwarves

* LICENSES-MAP: Add dwarves

* licenses.json: add dwarves

* dwarves: appease the linter (mostly)

Change fixes most of the linting issues. Leaving in the
	rm -Rf %{buildroot}
in %install. It is likely safe to remove but it was also part of the
initial imported spec so leaving it for now.

* dwarves: Fix license attribution in changelog

Cherry-picked from 7821a9e2ff576675ba3c4db5d4614571a621d1fc
with minor spec file modifications to expand cmake and ldconfig macros that are not present in 1.0.

Signed-off-by: Chris Co <chrco@microsoft.com>
###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local build

### CODE REVIEW

File: SPECS/dwarves/dwarves.spec [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+
+%install
+rm -Rf %{buildroot}
+DESTDIR="%{buildroot}" %__cmake --install .
+
+%post -p /sbin/ldconfig
+
+%postun -p /sbin/ldconfig
+
+%files
```
Comment (PawelWMS) [inline SPECS/dwarves/dwarves.spec:84]: Please include the license info with the `%license` macro somewhere.

File: SPECS/dwarves/dwarves.spec [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+%{_includedir}/dwarves/list.h
+%{_includedir}/dwarves/rbtree.h
+%{_includedir}/dwarves/pahole_strings.h
+%{_libdir}/%{libname}.so
+%{_libdir}/%{libname}_emit.so
+%{_libdir}/%{libname}_reorganize.so
+
+%changelog
+* Fri Aug 20 2021 Chris Co <chrco@microsoft.com> - 1.21-4
+- Initial CBL-Mariner import from Fedora 35 (License: MIT)
```
Comment (PawelWMS) [inline SPECS/dwarves/dwarves.spec:143]: Please check the license.

File: SPECS/dwarves/dwarves.spec [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+%description -n %{libname}%{libver}-devel
+Debugging information processing library development files.
+
+%prep
+%setup -q
+%patch1 -p1
+
+%build
+%cmake -DCMAKE_BUILD_TYPE=Release .
+%__cmake --build . %{?_smp_mflags} --verbose
```
Comment (PawelWMS) [inline SPECS/dwarves/dwarves.spec:74]: I think we can use the `%cmake_build` macro to take care of all the flags we need for our builds. That also makes global changes easier.

File: SPECS/dwarves/dwarves.spec [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+
+%install
+rm -Rf %{buildroot}
+DESTDIR="%{buildroot}" %__cmake --install .
```
Comment (PawelWMS) [inline SPECS/dwarves/dwarves.spec:78]: Similarly, I believe we should have a `%cmake_install` macro to replace this whole line.

File: SPECS/dwarves/dwarves.spec [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+%prep
+%setup -q
+%patch1 -p1
+
+%build
+%cmake -DCMAKE_BUILD_TYPE=Release .
+%__cmake --build . %{?_smp_mflags} --verbose
+
+%install
+rm -Rf %{buildroot}
```
Comment (PawelWMS) [inline SPECS/dwarves/dwarves.spec:77]: Nit: can be removed.

File: SPECS/dwarves/dwarves.spec [hunk 6]
```diff
@@ -0,0 +1,508 @@
...
+
+%package -n %{libname}%{libver}-devel
+Summary:        Debugging information library development files
+Requires:       %{libname}%{libver} = %{version}-%{release}
+
+%description -n %{libname}%{libver}-devel
+Debugging information processing library development files.
+
+%prep
+%setup -q
```
Comment (PawelWMS) [inline SPECS/dwarves/dwarves.spec:69]: Nit: `%autopatch -p1`? Not necessary, but makes adding more patches less error-prone.

Comment (christopherco) [reply SPECS/dwarves/dwarves.spec]: Fixed

File: SPECS/dwarves/dwarves.spec (hunk 1 above)
Comment (christopherco) [reply SPECS/dwarves/dwarves.spec:83]: Fixed

File: SPECS/dwarves/dwarves.spec (hunk 3 above)
Comment (christopherco) [reply SPECS/dwarves/dwarves.spec:73]: `%cmake_build` and `%cmake_install` macros are not available in 1.0

**Review by PawelWMS — APPROVED** [review]

**Review by anphel31 — APPROVED** [review]

**Pull request merged**